### PR TITLE
Added cucumberjs support and brand new unit tests

### PR DESCRIPTION
**Pull request by nikulkarni**

### CODE REVIEW

**Comment by nikulkarni** [comment]
@grawk Can you take a look at this?

**Comment by grawk** [comment]
Looking now

File: app/index.js [hunk 1]
```diff
@@ -200,29 +205,58 @@ var NemoGenerator = yeoman.generators.Base.extend({
...
     this.copy('test/functional/config/nemo-plugins.json', configDir + '/nemo-plugins.json');
     //locator dir
     this.mkdir(locatorDir);
-    this.copy('test/functional/locator/yhooreg.json', locatorDir + '/yhooreg.json');
+
 
     //report dir
     this.mkdir(reportDir);
     this.copy('test/functional/report/README.md', reportDir + '/README.md');
-    //spec dir
```
Comment (grawk) [inline app/index.js]: Why did you remove the yahoo example?

File: app/index.js [hunk 2]
```diff
@@ -149,37 +155,33 @@ var NemoGenerator = yeoman.generators.Base.extend({
...
+      if (!exists) {
+        var gruntContent = 'module.exports = function (grunt) { require(\'grunt-config-dir\')(grunt, {configDir: require(\'path\').resolve(\'tasks\')});};';
+        fs.writeFileSync('Gruntfile.js',gruntContent);
       }
-      if (data.match(/loopmocha/) !== null) {
-        //already here?
-        done();
-        return;
+    //let's update Gruntfile.js if possible
+    var api = require('gruntfile-api'),
```
Comment (grawk) [inline app/index.js]: Please move require to top of file and give more descriptive variable name like gruntfileAPI. Will help to clarify what's going on for folks coming to this fresh.

File: app/index.js [hunk 3]
```diff
@@ -200,29 +205,58 @@ var NemoGenerator = yeoman.generators.Base.extend({
       var newval = "<%=" + val + "%>";
       return newval;
     };
-    this.template('_loopmocha.js', taskDir + 'loopmocha.js');
+    if(this.testFramework==='mocha'){
```
Comment (grawk) [inline app/index.js]: Missing spaces around ===. I see this in a few places. Obviously stylistic only but pretty standard. If we were using jshint as a part of development, it would complain.. I'll file an issue that jshint should be a part of the build process going forward.

**Comment by grawk** [comment]
During unit tests I see some EACCES errors like this:

```bash
npm ERR! Darwin 12.5.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "--save-dev" "cucumber@^0.4.4" "nemo@^v0.2.0" "nemo-view@^v0.2.0" "nemo-drivex@^v0.1.0" "nemo-locatex@^v0.1.0" "nconf@~v0.6.7" "grunt-cucumberjs@^v0.4.1" "grunt-config-dir@^0.3.2"
npm ERR! node v0.10.29
npm ERR! npm  v2.1.17
npm ERR! path ../cucumber/bin/cucumber.js
npm ERR! code EACCES
npm ERR! errno 3

npm ERR! Error: EACCES, symlink '../cucumber/bin/cucumber.js'
npm ERR!  { [Error: EACCES, symlink '../cucumber/bin/cucumber.js'] errno: 3, code: 'EACCES', path: '../cucumber/bin/cucumber.js' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/medelman/src/_nemo/o/generator-nemo/temp/npm-debug.log
```

All tests still pass, but wondering if you see the same.

Also I've added a few comments to move stuff around or add some whitespace. If you can resolve those, and also squash all of this into a single commit, I'll merge then publish a new version.

Thanks for doing this work and I'm sincerely apologetic that I did not look at it sooner!

**Comment by nikulkarni** [comment]
@grawk Thanks for reviewing the PR. I understand your other commitments, I just wanted to post a friendly reminder earlier this week so this PR won't fall of your radar :)

I fixed formatting and moved the require to the top. As for the error you are seeing, I've seen it in the past  and it went away when I gave myself full access to .npm directory with this command.
```
sudo chown -R `whoami` ~/.npm
```

**Comment by grawk** [comment]
Published as v0.3.0
